src/Node/Nodes.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Node;
+
+use Twig\Attribute\YieldReady;
+
+/**
+ * Represents a list of nodes.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+#[YieldReady]
+class Nodes extends Node
```

Comment:
I think this class should be final, as custom nodes should not extend it (otherwise, we would have the same issue than with the usage of `Node` today)

Reply:
You're right, done here and for `EmptyNode` as well.